WARNING:root:You are using solc version 0.4.26. The supported version is 0.4.21.
INFO:root:Contract /sb/smart_billions.sol:SafeMath:
INFO:symExec:Running, please wait...
INFO:symExec:	============ Results ===========
INFO:symExec:	  EVM code coverage: 	 100.0%
INFO:symExec:	  Arithmetic bugs: 	 False
INFO:symExec:	  └> Overflow bugs: 	 False
INFO:symExec:	  └> Underflow bugs: 	 False
INFO:symExec:	  └> Division bugs: 	 False
INFO:symExec:	  └> Modulo bugs:   	 False
INFO:symExec:	  └> Truncation bugs: 	 False
INFO:symExec:	  └> Signedness bugs: 	 False
INFO:symExec:	  Callstack bug: 	 False
INFO:symExec:	  Concurrency bug: 	 False
INFO:symExec:	  Time dependency bug: 	 False
INFO:symExec:	  Reentrancy bug: 	 False
INFO:symExec:	 --- 0.0053870677948 seconds ---
INFO:symExec:	====== Analysis Completed ======
INFO:root:Contract /sb/smart_billions.sol:SmartBillions:
INFO:symExec:Running, please wait...
Traceback (most recent call last):
  File "/root/osiris/taintFlow.py", line 652, in perform_taint_analysis
    propagate_taint(taint, tainted_stack, tainted_memory, tainted_storage, instruction, current_stack, previous_block, current_block, next_blocks, arithmetic_errors, sha3_list, false_positives, strings)
  File "/root/osiris/taintFlow.py", line 418, in propagate_taint
    if arithmetic_error["instruction"].data_in[0] == 32 or arithmetic_error["instruction"].data_in[0] == 31:
  File "/root/osiris/z3/z3.py", line 302, in __nonzero__
    return self.__bool__()
  File "/root/osiris/z3/z3.py", line 310, in __bool__
    return self.arg(0).eq(self.arg(1))
  File "/root/osiris/z3/z3.py", line 920, in arg
    _z3_assert(is_app(self), "Z3 application expected")
  File "/root/osiris/z3/z3.py", line 89, in _z3_assert
    def _z3_assert(cond, msg):
  File "/root/osiris/symExec.py", line 2868, in handler
    raise Exception("timeout")
Exception: timeout
INFO:symExec:	============ Results ===========
INFO:symExec:	  EVM code coverage: 	 15.1%
INFO:symExec:	  Arithmetic bugs: 	 True
INFO:symExec:	  └> Overflow bugs: 	 True
/sb/smart_billions.sol:SmartBillions:380:13
wallets[msg.sender].balance += uint208(keepbalance)
^
/sb/smart_billions.sol:SmartBillions:339:47
investStart + (hashesSize * 5)
^
/sb/smart_billions.sol:SmartBillions:379:13
walletBalance += keepbalance
^
INFO:symExec:	  └> Underflow bugs: 	 True
/sb/smart_billions.sol:SmartBillions:344:26
block.number - hashFirst
^
/sb/smart_billions.sol:SmartBillions:345:29
dividends.length - 2
^
/sb/smart_billions.sol:SmartBillions:364:13
walletBalance -= balance
^
/sb/smart_billions.sol:SmartBillions:723:23
_block - hashFirst
^
INFO:symExec:	  └> Division bugs: 	 False
INFO:symExec:	  └> Modulo bugs:   	 False
INFO:symExec:	  └> Truncation bugs: 	 True
/sb/smart_billions.sol:SmartBillions:380:13
wallets[msg.sender].balance += uint208(keepbalance)
^
/sb/smart_billions.sol:SmartBillions:381:13
wallets[msg.sender].nextWithdrawBlock = uint32(block.number + 4 * 60 * 24 * 30)
^
/sb/smart_billions.sol:SmartBillions:486:15
hit & 0xF
^
INFO:symExec:	  └> Signedness bugs: 	 False
INFO:symExec:	  Callstack bug: 	 False
INFO:symExec:	  Concurrency bug: 	 True
Flow 1:
/sb/smart_billions.sol:SmartBillions:382:13
msg.sender.transfer(maxpay)
^
Flow 2:
/sb/smart_billions.sol:SmartBillions:372:13
msg.sender.transfer(_amount)
^
INFO:symExec:	  Time dependency bug: 	 False
INFO:symExec:	  Reentrancy bug: 	 False
INFO:symExec:	 --- 240.804929972 seconds ---
INFO:symExec:	====== Analysis Completed ======

  .oooooo.             o8o            o8o          
 d8P'  `Y8b            `"'            `"'          
888      888  .oooo.o oooo  oooo d8b oooo   .oooo.o
888      888 d88(  "8 `888  `888""8P `888  d88(  "8
888      888 `"Y88b.   888   888      888  `"Y88b. 
`88b    d88' o.  )88b  888   888      888  o.  )88b
 `Y8bood8P'  8""888P' o888o d888b    o888o 8""888P'



!!! SYMBOLIC EXECUTION TIMEOUT !!!
Unexpected error: <type 'exceptions.Exception'>
